service/history/historyEngine.go

```diff
@@ -2713,9 +2750,69 @@ func (e *historyEngineImpl) ReapplyEvents(
 				reapplyEvents,
 			); err != nil {
 				e.logger.Error("failed to re-apply stale events", tag.Error(err))
-				return nil, &workflow.InternalServiceError{Message: "unable to re-apply stale events"}
+				return emptyUpdateAction, &workflow.InternalServiceError{Message: "unable to re-apply stale events"}
 			}
 
 			return postActions, nil
 		})
 }
+
+func (e *historyEngineImpl) loadWorkflow(
+	ctx ctx.Context,
+	domainID string,
+	workflowID string,
+	runID string,
+) (workflowContext, error) {
+
+	context, release, err := e.historyCache.getOrCreateWorkflowExecution(
+		ctx,
+		domainID,
+		workflow.WorkflowExecution{
+			WorkflowId: common.StringPtr(workflowID),
+			RunId:      common.StringPtr(runID),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	mutableState, err := context.loadWorkflowExecution()
+	if err != nil {
+		release(err)
+		return nil, err
+	}
+
+	return newWorkflowContext(context, release, mutableState), nil
+}
+
+func (e *historyEngineImpl) loadCurrentWorkflow(
```

Comment:
this function should be called loadTargetWorkflow, not current workflow since run id is provided

Reply:
I renamed this to `loadWorkflow,` and previous loadWorkflow to `loadWorkflowHelper`, since API most likely won't need to call `loadWorkflowHelper` with the new change.